.changelog/5070.cfg.md

```diff
@@ -0,0 +1,103 @@
+Configuration changes
+
+The node now supports a configuration file that holds all the settings
+that were previously available via command-line options.
+The command-line options themselves have been removed in favor of
+the config file.
+
+The config file supports environment variable substitution using
+the `envsubst` package
+([usage information](https://github.com/a8m/envsubst#docs)).
+
+The following options have been removed:
+
+- `consensus.tendermint.mode` in favor of using the global node mode (`mode`).
+
+- `runtime.mode` in favor of using the global node mode (`mode`).
+
+- `worker.client.port` as it is no longer needed.
+
+- `worker.registration.force_register` as it is deprecated.
+
+The following options have been renamed:
+
+- `datadir` to `common.data_dir`.
+
+- `log.file` to `common.log.file`.
+
+- `log.format` to `common.log.format`.
+
+- `log.level` to `common.log.level`.  The log level is now a map of module
+to log level.  Use the `default` module to map the default log level.
+
+- `debug.rlimit` to `common.debug.rlimit`.
+
+- `debug.allow_root` to `common.debug.allow_root`.
+
+- `pprof.bind` to `pprof.bind_address`.
+
+- `consensus.tendermint.*` to `consensus.*`.
+
+- `consensus.tendermint.core.listen_address` to `consensus.listen_address`.
+
+- `consensus.tendermint.core.external_address` to
+`consensus.external_address`.
+
+- `consensus.tendermint.log.debug` to `consensus.log_debug`.
+
+- `consensus.tendermint.light_client.trust_period` to
+`consensus.state_sync.trust_period`.
+
+- `consensus.tendermint.seed.debug.disable_addr_book_from_genesis` to
+`consensus.debug.disable_addr_book_from_genesis`.
+
+- `consensus.tendermint.checkpointer.*` to `consensus.abci.checkpointer.*`.
+
+- `consensus.tendermint.sentry.upstream_address` to
+`consensus.sentry_upstream_addresses`.
+
+- `consensus.tendermint.upgrade.stop_delay` to
+`consensus.upgrade_stop_delay`.
+
+- `consensus.tendermint.supplementarysanity.*` to
+`consensus.supplementary_sanity.`.
+
+- `consensus.tendermint.p2p.persistent_peer` to
+`consensus.p2p.persistent_peers`.
+
+- `consensus.tendermint.p2p.unconditional_peer` to
+`consensus.p2p.unconditional_peers`.
+
+- `ias.proxy.address` to `ias.proxy_address`.
+
+- `ias.debug.skip_verify` to `ias.debug_skip_verify`.
+
+- `runtime.sandbox.binary` to `runtime.sandbox_binary`.
+
+- `runtime.sgx.loader` to `runtime.sgx_loader`.
+
+- `runtime.history.pruner.*` to `runtime.history_pruner.*`.
+
+- `worker.sentry.addresses` to `runtime.sentry_addresses`.
```

Comment:
Why is this not moved together with other sentry changes? I think it should just go under `sentry.addresses`?